### ISSUE TITLE
allow to use double quotes in commands passed via --exec (regression in v1.11)

### ISSUE THREAD

**Issue by arighi** [arighi/virtme-ng#27]
Commit 18680da ("virtme-ng: pass --exec command via /proc/cmdline") introduced a regression found by the mutter test case:
https://gitlab.gnome.org/GNOME/mutter/-/jobs/2917515

Passing the `--exec` command via `/proc/cmdline` directly, instead of using a temporary initramfs, is faster, but it has the downside that we don't support special characters anymore (such as double quotes for example).

To solve this regression and avoid breaking the old behavior we can pass the command as a base64-encoded string, that will be decoded in the guest by virtme-ng-init/virtme-init. The overhaed of the base64 encoding/decoding is negligible (definitely better than creating a temporary initramfs) and we can use any special character in the shell command.

**Comment by marcosps** [comment]
What, mutter is using virtme for their tests?? Very cool :+1:

**Comment by arighi** [comment]
> What, mutter is using virtme for their tests?? Very cool 👍

yep [1], and they switched to virtme-ng [2], we should probably run the same test as an autotest somehow to make sure we don't break it again in the future :sweat_smile: 

[1] https://gitlab.gnome.org/GNOME/mutter/-/blob/main/src/tests/kvm/virtme-run.sh
[2] https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/3029

**Comment by arighi** [comment]
Everything seems fixed in the mutter CI/CD chain with 94a3033161e22316a06aec6937f534292479885c applied, so we can close this.
